django/db/models/sql/query.py

```diff
@@ -1156,6 +1156,10 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
         if not arg:
             raise FieldError("Cannot parse keyword query %r" % arg)
         lookups, parts, reffed_expression = self.solve_lookup_type(arg)
+
+        if hasattr(reffed_expression, 'filterable') and not reffed_expression.filterable:
+            raise NotSupportedError(reffed_expression.__class__.__name__ + ' is disallowed in the filter-clause')
+
```

Comment:
Actually the `filterable` attribute would have a different meaning here, scratch that.